base/deprecated.jl

```diff
@@ -1499,4 +1499,12 @@ end
 # Calling promote_op is likely a bad idea, so deprecate its convenience wrapper promote_eltype_op
 @deprecate promote_eltype_op(op, As...) promote_op(op, map(eltype, As)...)
 
+# These conversions should not be defined, see #19896
+Base.convert{T<:Number}(::Type{T},x::Dates.Period) = error("conversion between Periods and Numbers is not defined")
+Base.convert{T<:Dates.Period}(::Type{T},x::Real) = error("conversion between Periods and Numbers is not defined")
+@deprecate convert{R<:Real}(::Type{R},x::Dates.DateTime) R(Dates.datetime2rata(x))
+@deprecate convert{R<:Real}(::Type{R},x::Dates.Date)     R(Dates.datetime2rata(x))
+@deprecate convert(::Type{Dates.DateTime}, x::Real) Dates.rata2datetime(x)
+@deprecate convert(::Type{Dates.Date}, x::Real)     Dates.Date(Dates.rata2datetime(x))
```

Comment:
Awesome, thanks for the help folks! I used these as a guideline to rework the modified tests at the bottom of `test/dates/conversions.jl`, here are the results:

```julia
julia> @test Dates.datetime2rata(a) == 63082368000000
Test Failed
  Expression: Dates.datetime2rata(a) == 63082368000000
   Evaluated: 730120 == 63082368000000
ERROR: There was an error during testing

julia> @test Dates.UTD(63082368000000) == a
Test Failed
  Expression: Dates.UTD(63082368000000) == a
   Evaluated: Base.Dates.UTInstant{Base.Dates.Day}(63082368000000 days) == 2000-01-01T00:00:00
ERROR: There was an error during testing

julia> @test Dates.UTD(63082368000000.0) == a
Test Failed
  Expression: Dates.UTD(6.3082368e13) == a
   Evaluated: Base.Dates.UTInstant{Base.Dates.Day}(63082368000000 days) == 2000-01-01T00:00:00
ERROR: There was an error during testing

julia> @test Dates.UTM(730120) == b
Test Failed
  Expression: Dates.UTM(730120) == b
   Evaluated: Base.Dates.UTInstant{Base.Dates.Millisecond}(730120 milliseconds) == 2000-01-01
ERROR: There was an error during testing

julia> @test Dates.UTM(730120.0) == b
Test Failed
  Expression: Dates.UTM(730120.0) == b
   Evaluated: Base.Dates.UTInstant{Base.Dates.Millisecond}(730120 milliseconds) == 2000-01-01
ERROR: There was an error during testing

julia> @test Dates.UTM(Int32(730120)) == b
Test Failed
  Expression: Dates.UTM(Int32(730120)) == b
   Evaluated: Base.Dates.UTInstant{Base.Dates.Millisecond}(730120 milliseconds) == 2000-01-01
ERROR: There was an error during testing
```
I may have screwed some of those up, but it appears we aren't there yet.